backend/pkg/kafka/deserializer_integration_test.go

```diff
@@ -0,0 +1,972 @@
+}
+
+func TestSuite(t *testing.T) {
+	suite.Run(t, &KafkaIntegrationTestSuite{})
+}
+
+func (s *KafkaIntegrationTestSuite) createBaseConfig() config.Config {
+	cfg := config.Config{}
+	cfg.SetDefaults()
+	cfg.MetricsNamespace = testutil.MetricNameForTest("deserializer")
+	cfg.Kafka.Brokers = []string{s.seedBroker}
+
+	return cfg
+}
+
+func (s *KafkaIntegrationTestSuite) consumerClientForTopic(topicName string) *kgo.Client {
+	t := s.T()
+	require := require.New(t)
+
+	cl, err := kgo.NewClient(
+		kgo.SeedBrokers(s.seedBroker),
+		kgo.ConsumeTopics(topicName),
+		kgo.ConsumeResetOffset(kgo.NewOffset().AtStart()),
+	)
+	require.NoError(err)
+
+	return cl
+}
+
+func (s *KafkaIntegrationTestSuite) SetupSuite() {
+	t := s.T()
+	require := require.New(t)
+
+	ctx := context.Background()
+
+	// Redpanda container
+	exposedKafkaPort := rand.Intn(50000) + 10000 //nolint:gosec // We can use weak random numbers for ports in tests.
+	// externalKafkaPort := rand.Intn(50000) + 10000 //nolint:gosec // We can use weak random numbers for ports in tests.
+	// exposedKafkaAdminPort := rand.Intn(50000) + 10000 //nolint:gosec // We can use weak random numbers for ports in tests.
+	schemaRegistryPort := rand.Intn(50000) + 10000 //nolint:gosec // We can use weak random numbers for ports in tests.
+
+	redpandaContainer, err := runRedpandaForKafkaTests(ctx, exposedKafkaPort, schemaRegistryPort)
+	require.NoError(err)
+
+	s.redpandaContainer = redpandaContainer
+
+	seedBroker, err := getMappedHostPort(ctx, redpandaContainer, nat.Port(strconv.FormatInt(int64(exposedKafkaPort), 10)+"/tcp"))
+	require.NoError(err)
+
+	s.seedBroker = seedBroker
+	s.kafkaClient, s.kafkaAdminClient = testutil.CreateClients(t, []string{seedBroker})
+
+	registryAddr, err := getMappedHostPort(ctx, redpandaContainer, nat.Port(strconv.FormatInt(int64(schemaRegistryPort), 10)+"/tcp"))
+	require.NoError(err)
+	s.registryAddress = registryAddr
+
+	logCfg := zap.NewDevelopmentConfig()
+	logCfg.Level = zap.NewAtomicLevelAt(zap.InfoLevel)
+	log, err := logCfg.Build()
+	require.NoError(err)
+
+	s.log = log
+}
+
+func (s *KafkaIntegrationTestSuite) TearDownSuite() {
+	t := s.T()
+	assert := require.New(t)
+
+	assert.NoError(s.redpandaContainer.Terminate(context.Background()))
+}
+
+func (s *KafkaIntegrationTestSuite) TestDeserializeRecord() {
+	t := s.T()
+
+	require := require.New(t)
+	assert := assert.New(t)
+
+	ctx := context.Background()
+
+	t.Run("plain JSON", func(t *testing.T) {
+		testTopicName := testutil.TopicNameForTest("deserializer_plain_json")
+		_, err := s.kafkaAdminClient.CreateTopic(ctx, 1, 1, nil, testTopicName)
+		require.NoError(err)
+
+		defer func() {
+			_, err := s.kafkaAdminClient.DeleteTopics(ctx, testTopicName)
+			assert.NoError(err)
+		}()
+
+		cfg := s.createBaseConfig()
+		cfg.Kafka.Protobuf.Enabled = true
+		cfg.Kafka.Protobuf.SchemaRegistry.Enabled = true
+		cfg.Kafka.Schema.Enabled = true
+		cfg.Kafka.Schema.URLs = []string{"http://" + s.registryAddress}
```

Comment:
Should this be part of the base config? These are not necessarily needed for the plain JSON case. Possibly another test case to check if JSON deserialization works with both schema registry enabled/disabled if that was intended I'd recommend a comment here.